File: project/src/services/PaiementService.js
import axios from "axios";

const REST_API_BASE_URL='http://localhost:8080/paiement';

export const updatePaiement=(idPaiement,paiement) => axios.put(REST_API_BASE_URL+"/"+idPaiement,paiement);
export const listPaiement=() => axios.get(REST_API_BASE_URL);
export const getRevenu=() => axios.get(REST_API_BASE_URL+"/revenu");
export const getRevenuParMois=(year) => axios.get(REST_API_BASE_URL+"/revenu/"+year,year);

File: BackCarRental/src/main/java/net/codejava/BackCarRental/service/PaiementService.java
package net.codejava.BackCarRental.service;

import net.codejava.BackCarRental.dto.PaiementDTO;
import net.codejava.BackCarRental.model.Paiement;

import java.util.List;

public interface PaiementService {
    List<PaiementDTO> getAllPaiements();
    PaiementDTO getPaiementById(String paiementId);
    PaiementDTO updatePaiement(String paiementId, PaiementDTO updatedPaiement);
    double getRevenu();
    List<Object[]> getRevenuParMois(int year);
    PaiementDTO createPaiement(PaiementDTO paiementDTO);
}
